infra/build/functions/build_project.py

```diff
@@ -337,6 +348,12 @@ def get_build_steps(  # pylint: disable=too-many-locals, too-many-statements, to
               ],
           })
 
+          # Report the build failure if it happened.
+          build_steps.append({
```

Comment:
Eh, I don't love what this will do to the UI where the step where the failure actually occurs will have a green checkmark.

Reply:
Yeah, but I couldn't come up with a better way. This is only for the experiment runs where we want to upload the compile logs separately.